### PR TITLE
Fix make verify

### PR DESCRIPTION
**Pull request by philoserf**
**What this PR does / why we need it**:

`make verify` did not work on the existing code base.

**Which issue this PR fixes** :

NONE

**Special notes for your reviewer**:

Checkout master and try `make verify`

```
...
Verifying ./bin/../bin/checks/verify-golint.sh
FAILED   ./bin/../bin/checks/verify-golint.sh	0s
...
```

checkout this PR and try `make verify` all tests pass.

other `make` tasks also continue to work as expected.

---
This change is [Reviewable](https://reviewable.io/reviews/samsung-cnct/kraken/199)

### CODE REVIEW

File: bin/common.sh [hunk 1]
```diff
@@ -31,11 +31,5 @@ packages() {
...
 
 valid_go_files() {
-  find . -not \( \
-      \( \
-        -wholename './output' \
-        -o -wholename '*/vendor/*' \
-        -o -wholename '*/bindata.go' \
-      \) -prune \
-    \) -name '*.go'
+  git ls-files "**/*.go" | grep -v -e "vendor" -e "cmd/bindata.go"
```
Comment (alejandroEsc) [inline bin/common.sh]: I would prefer to leave the command to be linux specific. Not sure what are the thoughts on this. could we just exclude the main.go command only? The other thing to keep in mind is there are some checks that will pass and not make the same complaint about not being in the same folder.

File: cmd/cluster_helper.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ const (
 
 func preRunGetClusterConfig(cmd *cobra.Command, args []string) error {
 	if ClusterConfigPath == "" {
-		return fmt.Errorf("please pass a valid kraken config file.")
+		return fmt.Errorf("please pass a valid kraken config file")
```
Comment (alejandroEsc) [inline cmd/cluster_helper.go]: great catch!

File: cmd/docker_client_config.go [hunk 3]
```diff
@@ -5,9 +5,10 @@ import (
 	"strconv"
 
 	"crypto/tls"
+	"path/filepath"
+
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/tlsconfig"
-	"path/filepath"
```
Comment (alejandroEsc) [inline cmd/docker_client_config.go]: was this change on purpose? If not could you please revert this.

File: cmd/release_helper.go [hunk 4]
```diff
@@ -41,7 +42,7 @@ func compareReleases(versionA string, versionB string) (int, error) {
 }
 
 func krakenLibTagToSemver(tag string) string {
-	if tag == VERSION_LATEST {
+	if tag == latestVersion {
 		return tag
 	}
```
Comment (alejandroEsc) [inline cmd/release_helper.go]: Another great catch, thank you for fixing that.

File: cmd/docker_client_config.go (hunk 3 above)
Comment (philoserf) [reply cmd/docker_client_config.go]: This is the Golang community convention. The commonly used goimports does this for you.

File: bin/common.sh (hunk 1 above)
Comment (philoserf) [reply bin/common.sh]: This isn’t Linux. This it git. It is a common Golang Makefile idiom. I am happy to find some _in the wild_ examples from prominent Golang projects.

File: cmd/release_helper.go (hunk 4 above)
Comment (philoserf) [reply cmd/release_helper.go]: Not really a great catch for me. While make verify didn’t work a well crafted golint did.

File: cmd/cluster_helper.go (hunk 2 above)
Comment (philoserf) [reply cmd/cluster_helper.go]: Not really a great catch for me. While make verify didn’t work a well crafted golint did.

File: bin/common.sh (hunk 1 above)
Comment (philoserf) [reply bin/common.sh]: This excludes files we do not own
- vendor
And a generated file that doesn’t follow linting standards
- bindata.go

main.go is still checked

We should consider adding a make command or at least docs in generation the bindata.go too. But I leave that for the rewrite that would accompany removing the gox tool.

Comment (philoserf) [reply bin/common.sh]: I am wrong. The shell glob `**/*.go` does skip `main.go`

**Comment by alejandroEsc** [comment]
@philoserf could you please add that well crafted golint to the make verify?

**Comment by philoserf** [comment]
The is what line 24 of bin/common.sh does.

**Comment by alejandroEsc** [comment]
@philoserf sorry are you sure about that?

**Comment by alejandroEsc** [comment]
@philoserf please i must be looking at the wrong file, let me know if thats what you meant. Ready to merge this in today if possible.

File: bin/common.sh (hunk 1 above)
Comment (coffeepac) [reply bin/common.sh]: can you please modify the `git ls-files` command to include `main.go`?  its a very small file but just in case.

**Comment by philoserf** [comment]
I think this is ready. I will share my one liner to run individual verify sub tasks with out of band.

File: cmd/docker_client_config.go (hunk 3 above)
Comment (alejandroEsc) [reply cmd/docker_client_config.go]: ahh ok, misread where this was changed. gotcha.

**Comment by philoserf** [comment]
Hold. I have one minor update coming.

**Comment by philoserf** [comment]
Here is the one-liner to check subtasks. In this example I check each file with golint.

`git ls-files "**/*.go" "*.go" | grep -v -e "vendor" -e "cmd/bindata.go"|xargs -I {} golint {}`

**Comment by philoserf** [comment]
All clear. Tests Pass. Verify works.
